src/sagemaker/workflow/step_collections.py

```diff
@@ -130,13 +130,10 @@ def __init__(
             steps.append(repack_model_step)
             model_data = repack_model_step.properties.ModelArtifacts.S3ModelArtifacts
 
-        # remove kwargs consumed by model repacking step
-        kwargs.pop("entry_point", None)
-        kwargs.pop("source_dir", None)
-        kwargs.pop("dependencies", None)
-        kwargs.pop("output_kms_key", None)
+            # remove kwargs consumed by model repacking step
+            kwargs.pop("output_kms_key", None)
 
-        if model is not None:
+        elif model is not None:
```

Comment:
Hi @sreedes, can you clarify why this change to an `elif` is necessary? It seems like checking for existence of `entry_point` and `model` are two independent things, and we would likely want to check both?

Reply:
Both of them cannot appear together, the customer could either send it through model or directly through RegisterModelStep(existing mechanism), we would recommend them to use it through Model, since it comes packed with framework information as well, hence the else.

Reply:
Thanks for clarifying!